From 670cce502aadf86b5b5d78059e5474e6171919f3 Mon Sep 17 00:00:00 2001
From: nekral-guest <nekral-guest@5a98b0ae-9ef6-0310-add3-de5d479b70d7>
Date: Sat, 30 Aug 2008 18:30:58 +0000
Subject: [PATCH] * lib/groupio.h, lib/prototypes.h, lib/pwio.h, lib/sgetgrent.c:
 	Include <sys/types.h> before <pwd.h> and <grp.h>. It is necessary
 	for the definition of uid_t and gid_t.
 	* lib/pwmem.c: do not include <pwd.h>, "pwio.h" is sufficient
 	here.

git-svn-id: svn://svn.debian.org/pkg-shadow/upstream/trunk@2297 5a98b0ae-9ef6-0310-add3-de5d479b70d7
---
 lib/groupio.h    |    1 +
 lib/prototypes.h |    1 +
 lib/pwio.h       |    2 ++
 lib/pwmem.c      |    3 +--
 lib/sgetgrent.c  |    1 +
 6 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/lib/groupio.h b/lib/groupio.h
index 9f2984c..d229845 100644
--- a/lib/groupio.h
+++ b/lib/groupio.h
@@ -35,6 +35,7 @@
 #ifndef _GROUPIO_H
 #define _GROUPIO_H
 
+#include <sys/types.h>
 #include <grp.h>
 
 extern int gr_close (void);
diff --git a/lib/prototypes.h b/lib/prototypes.h
index f1ffc50..feeedc4 100644
--- a/lib/prototypes.h
+++ b/lib/prototypes.h
@@ -48,6 +48,7 @@
 #else
 #include <utmp.h>
 #endif
+#include <sys/types.h>
 #include <pwd.h>
 #include <grp.h>
 #include <shadow.h>
diff --git a/lib/pwio.h b/lib/pwio.h
index 28f8bbd..52c7bf3 100644
--- a/lib/pwio.h
+++ b/lib/pwio.h
@@ -35,7 +35,9 @@
 #ifndef _PWIO_H
 #define _PWIO_H
 
+#include <sys/types.h>
 #include <pwd.h>
+
 extern int pw_close (void);
 extern const struct passwd *pw_locate (const char *name);
 extern const struct passwd *pw_locate_uid (uid_t uid);
diff --git a/lib/pwmem.c b/lib/pwmem.c
index 95a6137..84dee3d 100644
--- a/lib/pwmem.c
+++ b/lib/pwmem.c
@@ -35,10 +35,9 @@
 
 #ident "$Id: shadow-4.1.2.2-id-types.patch,v 1.1 2009/03/15 04:56:23 vapier Exp $"
 
+#include <stdio.h>
 #include "prototypes.h"
 #include "defines.h"
-#include <pwd.h>
-#include <stdio.h>
 #include "pwio.h"
 
 struct passwd *__pw_dup (const struct passwd *pwent)
diff --git a/lib/sgetgrent.c b/lib/sgetgrent.c
index 6f090aa..186ee40 100644
--- a/lib/sgetgrent.c
+++ b/lib/sgetgrent.c
@@ -35,6 +35,7 @@
 #ident "$Id: shadow-4.1.2.2-id-types.patch,v 1.1 2009/03/15 04:56:23 vapier Exp $"
 
 #include <stdio.h>
+#include <sys/types.h>
 #include <grp.h>
 #include "defines.h"
 #include "prototypes.h"
-- 
1.6.2

